packages/simcore-sdk/tests/unit/test_node_ports_v2_port.py

```diff
@@ -77,18 +78,51 @@ def e_tag() -> str:
 ##################### FIXTURES
 
 
-@pytest.fixture
-def this_node_file(tmp_path: Path) -> Path:
-    file_path = this_node_file_name()
-    file_path.write_text("some dummy data")
+@pytest.fixture(
+    params=[
+        True,
+        False,
+    ]
+)
+def is_symlink(request) -> bool:
+    return request.param
+
+
+@pytest.fixture()
+def this_node_file(is_symlink: bool, tmp_path: Path) -> Iterator[Path]:
+    """return either a symlink or a file with the same name"""
+
+    file_name = this_node_file_name()
+
+    symlink_path: Optional[Path] = None
+    return_path: Optional[Path] = None
+
+    if is_symlink:
+        symlink_path = file_name
+        file_path = file_name.parent / f"source_{file_name.name}"
+        file_path.write_text("some dummy data")
```

Comment:
you might want to separate that fixture in 2, you might even just use the tempfile fixture for that...
in case it's a symlink you return the symlink, and if not you return the file itself.

Reply:
Think I managed to make it more readable.